### ISSUE TITLE
Empty key in psr-4 breaks composer

### ISSUE THREAD

**Issue by vtek21** [briannesbitt/Carbon#1374]
By using an empty key in the psr-4 autoload section in the composer.json, the key get's replaced with the `_empty_` string in php (like in https://bugs.php.net/bug.php?id=46600).

This behavior causes composer to fail generating optimized autoloader files with the following error:

```
Generating optimized autoload files

  [InvalidArgumentException]
  psr-4 namespaces must end with a namespace separator, '_empty_' does not, use '_empty_\'.
```

Error above gets triggered because this part is in the installed.json

```
        "autoload": {
            "psr-4": {
                "_empty_": "src/"
            }
        }
```

This problem is a bit hard to reproduce. But if you delete your composer.lock and the composer folder from your vendor - and after that perform a composer install -  it should behave the way I mentioned above.

**Comment by kylekatarnls** [comment]
Before to hurry in quick-fix, I would mention our constraints:

This have to continue to work from PHP 5.3 to PHP 7.2:
https://github.com/briannesbitt/Carbon/pull/1207

If the `JsonSerialize` polyfill is not able to load (not in any namespace), then we cannot go further.

Also I would like to mention we strictly followed the Composer documentation:
https://getcomposer.org/doc/04-schema.md#psr-4

As it's documented here, it's widely used and if Composer is supposed to support this syntax, we're not supposed to work around but report a bug in Composer and find the better solution with them.

Last, I would have a systematic way to reproduce this bug to exactly understand in which case it happens.

**Comment by kylekatarnls** [comment]
We still waiting for details to handle this. Composer version, PHP version, command line used a minima. I took a project having Carbon in its dependencies, I removed composer.lock and vendor/composer, I ran `composer install`, then redo the same with `composer install -o`.

I tried with a big project with many dependencies and with a project that only includes Carbon. I could not reproduce it. So I close this for now.
